src/pages/ProductList/ProductList.js

```diff
@@ -52,136 +51,231 @@ class ProductList extends Component {
+          );
+        }
+      );
+
+      // 10/28 수요일에 백엔드 분들과 맞춰볼 예정입니다.
+      // fetch(`http://`, {
+      //   method: "GET",
+      //   body: JSON.stringify({
+      //     endpoint: `/shop?category=All&page=1`
+      //   })
+      // })
+      //   .then(res => res.json())
+      //   .then(result =>
+      // console.log("result", result)
+      // this.setState({
+      //   productsByCategory: result.data
+      // }))
+      // result.data는 임의로 입력한 키값입니다. (백엔드 맞춰보고 수정 예정)
+    }
+
+    const filteredProducts = allProducts.filter((product) => {
+      return product.category === category;
+    });
+    this.setState(
+      {
+        isPageFooterVisible: filteredProducts.length < 13,
         productsByCategory: [...filteredProducts],
         productsByPage: [...filteredProducts],
-        isPageFooterVisible: false}, () => {this.props.history.push(`/product-category/${categoryName[category]}/`)})
-  }
+      },
+      () => {
+        this.props.history.push(`/product?category=${categoryName[category]}`);
+      }
+    );
+
+    // 10/28 수요일에 백엔드 분들과 맞춰볼 예정입니다.
+    // fetch(`http://`, {
+    //   method: "GET",
+    //   body: JSON.stringify({
+    //     endpoint: `/product?category=${categoryName[category]}`
+    //   })
+    // })
+    //   .then(res => res.json())
+    //   .then(result => console.log("endpoint", result))
+    //   .then(result =>
+    //     console.log("result", result)
+    //       this.setState({
+    //       productsByCategory: result.data,
+    //       productsByPage: result.data,
+    // }))
+    // result.data는 임의로 입력한 키값입니다. (백엔드 맞춰보고 수정 예정)
+  };
 
   goToProductDetail = (id) => {
-    console.log(this.props.history)
-    this.props.history.push(`/shop/${id}`)
-  }
+    // fetch(`API/shop/${id}`, {method: "GET"})
+    //     .then(res => res.json())
+    //     .then(result => console.log(result))
+
+    this.props.history.push(`/shop/${id}`);
+  };
+
+  goToCartPage = () => {
+    // fetch(`API/cart`, {method: "GET"})
+    //     .then(res => res.json())
+    //     .then(result => console.log(result))
+
+    this.props.history.push(`/cart`);
+  };
 
   filterByPage = (num) => {
     const { productsByCategory } = this.state;
-    const slicePageIdx = num === 1 ? [0, 12] : num === 2 ? [12, 24] : num === 3 ? [24, 36] : [36];
-    const prevBtnBool = num === 1 ? false : num === 2 || num === 3 ? true : true;
-    const nextBtnBool = num === 1 ? true : num === 2 || num === 3 ? true : false;
-
-    this.setState({
-      productsByPage : productsByCategory.slice(...slicePageIdx),
-      mappingPage : true,
-      isPrevBtnVisible : prevBtnBool,
-      isNextBtnVisible : nextBtnBool
-    })
-  }
+    const slicePageIdx =
+      num === 1 ? [0, 12] : num === 2 ? [12, 24] : num === 3 ? [24, 36] : [36];
+    const prevBtnBool =
+      num === 1 ? false : num === 2 || num === 3 ? true : true;
+    const nextBtnBool =
+      num === 1 ? true : num === 2 || num === 3 ? true : false;
```

Comment:
인자로 받는 num 이 1, 2, 3에 해당할 경우 boolean으로 결과를 도출하는 것이네요! 
```suggestion
    const nextBtnBool =
      num === 1 || num === 2 || num === 3 ? true : false;
```
이렇게 중간 true를 생략해줄 수 도 있을 것 같습니다 :)

Reply:
앗 이런 실수를...... 다음에는 꼼꼼히 살펴서 중복은 모두 제거하도록 하겠습니다!

해당 부분은 백엔드와 맞춰보면서 pagination으로 대체되었어요 :)